request_provider/__brick__/request_state.dart

```diff
@@ -3,20 +3,20 @@ import 'package:freezed_annotation/freezed_annotation.dart';
 part 'request_state.freezed.dart';
 
 @freezed
-abstract class RequestState<Value, Error extends Exception> with _$RequestState<Value, Error> {
-  const factory RequestState.initial() = RequestStateInitial<Value, Error>;
+abstract class RequestState<Value> with _$RequestState<Value> {
+  const factory RequestState.initial() = RequestStateInitial<Value>;
 
-  const factory RequestState.loading({Value? resultMaybe}) = RequestStateLoading<Value, Error>;
+  const factory RequestState.loading({Value? resultMaybe}) = RequestStateLoading<Value>;
 
-  const factory RequestState.success(Value result) = RequestStateSuccess<Value, Error>;
+  const factory RequestState.success(Value result) = RequestStateSuccess<Value>;
 
-  const factory RequestState.failure(Error error) = RequestStateFailure<Value, Error>;
+  const factory RequestState.failure(Object error) = RequestStateFailure<Value>;
 }
 
-extension IsLoading<Value, Error extends Exception> on RequestState<Value, Error> {
+extension IsLoading<Value> on RequestState<Value> {
   bool get isLoading => maybeMap(orElse: () => false, loading: (value) => true);
 
   Value? get value => maybeWhen(orElse: () => null, success: (result) => result);
 
-  Exception? get error => maybeWhen(orElse: () => null, failure: (error) => error);
+  Object? get error => maybeWhen(orElse: () => null, failure: (error) => error);
```

Comment:
Change the extension name to something more general, something like `RequestStateExtension`.
Add `isInitial`, `isSuccess` and `isFailure`, just like we have `isLoading`.